src/Orleans.Runtime/Hosting/EndpointOptions.cs

```diff
@@ -8,25 +10,97 @@ namespace Orleans.Hosting
     public class EndpointOptions
     { 
         /// <summary>
-        /// The external IP address or host name used for clustering.
-        /// </summary>
-        public string HostNameOrIPAddress { get; set; }
-
-        /// <summary>
-        /// The IP address used for clustering. Will be inferred from <see cref="HostNameOrIPAddress"/> if this is not directly specified.
+        /// The IP address used for clustering.
         /// </summary>
         public IPAddress IPAddress { get; set; }
 
         /// <summary>
         /// The port this silo uses for silo-to-silo communication.
         /// </summary>
-        public int Port { get; set; }
```

Comment:
For ports, do we want to have defaults? We've been documenting that 11111/30000 are the default ports we use, even though I don't see that we actually do that. I see three options here:
* Set them to 11111/30000 be default
* Set tthem to randomly picked available ports
* Do nothing, and fail to start a silo if the ports aren't specified

Reply:
- If we fail we should do it with a proper exception message. We could add the required components to configure in the SiloHostBuilder
- I like the idea of random ports, if people want to run multiple silos on the same machine, it can really help them, but...
- Since it is not the only required core components for Orleans to run (we always need membership for example), could we have a default builder/methods that configure all required components with default options? This way all default will be in one place

Reply:
I'd down vote the random port suggestion: it's more important to be deterministic, than supporting the multi-silo deployment without extra configuration. I'd say it would be more common to use containers for multi-silo deployments, where in fact random ports can be an issue (as it is in many locked-down-network deployments where the firewall needs to be explicitly configured, which is the majority of cases).

I'd prefer fixed port, but it would be not terrible if we throw a clear exception when unassigned, the question is what do we gain? (except that we silently bind even if there is another silo listening on the same port, behavior that I disagree with and shouldn't be the default).

Reply:
My vote is for defaulting to 11111/30000.

Reply:
Anyway I don't think we should fix this in the PR?